IngressMaxFields.user.js

```diff
@@ -133,47 +132,152 @@ function wrapper() {
+                            </td>
+                            <td width='50%'>
+                                <input type='number' class='num_agents' name='num_agents' value='1' min='1' required>
+                            </td>
+                        </tr>
+                        <tr>
+                            <td>
+                                Use Google Maps?
+                            </td>
+                            <td>
+                                <input type='checkbox' name='useGoogle' value='YES' checked>
+                            </td>
+                        </tr>
+                        <tr>
+                            <td>
+                                Color scheme
+                            </td>
+                            <td>
+                                <input type='radio' name='color' value='ENL' checked>ENL</input>
+                                <input type='radio' name='color' value='RES'>RES</input>
+                            </td>
+                        </tr>
+                    </table>
+                </div>
+                <div id='submit'>
+                    <table>
+                        <tr>
+                            <td>
+                                Email:
+                            </td>
+                            <td>
+                                <input type='email' name='email' placeholder='(optional)'>
+                            </td>
+                        </tr>
+                        <tr>
+                            <td></td>
+                            <td>
+                                <input type='submit' class='submit' name='submit' value='Submit!'>
+                            </td>
+                        </tr>
+                    </table>
+                </div>
+            </div>
+        </form>
+        `;
         var dia = window.dialog({
-                title: "www.ingress-maxfield.com: Field your future",
-                html: data
-            }).parent();
+            title: "www.ingress-maxfield.com: Field your future",
+            html: data
+        }).parent();
         $(".ui-dialog-buttonpane", dia).remove();
         dia.css("width", "600px").css("top", ($(window).height() - dia.height()) / 2).css("left", ($(window).width() - dia.width()) / 2);
         return dia;
     };
 
     self.gen = function gen() {
         var o = self.checkPortals(window.portals);
-        var dialog = self.showDialog(o.list);
+        var bookmarks = self.checkBookmarks();
+        var dialog = self.showDialog(o.list, bookmarks);
         return dialog;
     };
 
+    // Return a list of portal bookmark folders
+    self.checkBookmarks = function checkBookmarks() {
+        if (!window.plugin.bookmarks) {
+            return null;
+        }
+
+        var portalsList = JSON.parse(localStorage['plugin-bookmarks']);
+
+        var res = {};
+        for (var folder in portalsList.portals) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype.](https://www.codacy.com/app/stefan-nygren/IITC-Ingress-Maxfields-Exporter/file/2651000314/issues/source?bid=3039347&fileBranchId=3324533#l275)